File: 10-10-2024_Max_distance_between_same_elements.java
class Solution {
    public int maxDistance(int[] arr) {
        // Code here
        int n = arr.length;
        int maxD = -1; 
        for (int i = 0; i < n - 1; i++) { 
            for (int j = i + 1; j < n; j++) { 
                // check if two elements are equal 
                if (arr[i] == arr[j]) { 
                    // if yes then calculate the distance 
                    // and update maxD 
                    int temp = Math.abs(j - i); 
                    maxD = maxD > temp ? maxD : temp; 
                } 
            } 
        } 
         // return maximum distance 
        return maxD; 
    }
}

File: 10-10-2024_Max_distance_between_same_elements.py
class Solution:

    def maxDistance(self, arr):
        n = len(arr)
        mp = {}
        max_dist = 0
        for i in range(n):
            if arr[i] not in mp:
                mp[arr[i]] = i
            else:
                max_dist = max(max_dist, i - mp[arr[i]])
        return max_dist
